helper/resource/error.go

```diff
@@ -0,0 +1,21 @@
+package resource
+
+type NotFoundError struct {
```

Comment:
It would be good to name error types consistently between here & #5205. I don't have a strong preference either way.